applications/sandbox/systems/physics_test_system.cpp

```diff
@@ -0,0 +1,729 @@
+        suzzaneH = rendering::ModelCache::create_model("suzanne", "assets://models/suzanne.glb"_view);
+        teapotH = rendering::ModelCache::create_model("sah", "assets://models/gnomecentered.obj"_view);
+
+        #pragma endregion
+
+        #pragma region Input binding
+
+        app::InputSystem::createBinding<extendedPhysicsContinue>(app::inputmap::method::M);
+        app::InputSystem::createBinding<nextPhysicsTimeStepContinue>(app::inputmap::method::N);
+        app::InputSystem::createBinding<QHULL>(app::inputmap::method::Q);
+        app::InputSystem::createBinding<AddRigidbody>(app::inputmap::method::R);
+        app::InputSystem::createBinding<SpawnRandomHullOnCameraLoc>(app::inputmap::method::F);
+        app::InputSystem::createBinding<SpawnHullActive>(app::inputmap::method::P);
+
+        #pragma endregion
+        
+        #pragma region Function binding
+        bindToEvent<extendedPhysicsContinue, &PhysicsTestSystem::extendedContinuePhysics>();
+        bindToEvent<nextPhysicsTimeStepContinue, &PhysicsTestSystem::OneTimeContinuePhysics>();
+       
+        bindToEvent<QHULL, &PhysicsTestSystem::quickHullStep>();
+        bindToEvent<AddRigidbody, &PhysicsTestSystem::AddRigidbodyToQuickhulls>();
+        bindToEvent< SpawnRandomHullOnCameraLoc, &PhysicsTestSystem::spawnRandomConvexHullOnCameraLocation>();
+        bindToEvent< SpawnHullActive, &PhysicsTestSystem::ActivateSpawnRandomHull>();
+
+        #pragma endregion
+
+        auto lightshader = rendering::ShaderCache::create_shader("light", "assets://shaders/light.shs"_view);
+        directionalLightMH = rendering::MaterialCache::create_material("directional light", lightshader);
+        directionalLightMH.set_param("color", math::color(1, 1, 0.8f));
+        directionalLightMH.set_param("intensity", 1.f);
+
+        {
+            auto sun = createEntity();
+            sun.add_component<rendering::mesh_renderable>(mesh_filter(directionalLightH.get_mesh()), rendering::mesh_renderer(directionalLightMH));
+            sun.add_component<rendering::light>(rendering::light::directional(math::color(1, 1, 0.8f), 10.f));
+            sun.add_component<transform>(position(10, 10, 10), rotation::lookat(math::vec3(1, 1, -1), math::vec3::zero), scale());
+        }
+
+        createProcess<&PhysicsTestSystem::colliderDraw>("Physics",0.02f);
+
+        quickhullTestScene();
+    }
+
+    void PhysicsTestSystem::colliderDraw(time::span dt)
+    {
+        ecs::filter<ObjectToFollow> objectToFollowQuery;
+
+        for (auto ent : objectToFollowQuery)
+        {
+            auto& objToFollow = ent.get_component<ObjectToFollow>().get();
+
+            auto [posH,rotH,scaleH] = ent.get_component<transform>();
+            auto [posH2, rotH2, scaleH2] = objToFollow.ent.get_component<transform>();
+
+            posH = posH2.get();
+            rotH = rotH2.get();
+            scaleH = scaleH2.get();
```

Comment:
I get that this is just a test system, so i don't rly care tbh. but keep in mind that any of these handles might be invalid. and that will cause an exception.